eip-6170.md

```diff
@@ -0,0 +1,79 @@
+
+## Motivation
+
+Cross-chain messaging protocols lack standardization leading to diverse implementation detail. Some examples include Layerzero, Hyperlane & Wormhole, having their own sets of implementations. This makes integration difficult at the aggregator or plugin layer for protocols that must conform to any standards and forces each protocol to implement its adapter.
+
+Even chain native arbitrary messaging protocols like Matic State Tunnel has its own way of implementation.
+
+The lack of standardization not only makes integration difficult but also is error prone as developers need to understand the operation of the messaging protocol at its core.
+
+A standard for cross-chain messaging will lower integration efforts for cross-chain messaging while creating more robust & secure cross-chain application development.
+
+To the worst, each messaging protocol has its chain ids (or) chain identification mechanism. However, this mechanism is used as they support non-EVM chains, which makes integration a nightmare.
+
+## Specification
+
+The keywords "MUST", "MUST NOT", "REQUIRED", "SHALL", "SHALL NOT", "SHOULD", "SHOULD NOT", "RECOMMENDED", "MAY", and "OPTIONAL" in this document are to be interpreted as described in RFC 2119.
+
+### Methods
+
+sendMessage
+Dispatch message/state to `receiver` on a predetermined `chain id`.
+MUST emit `MessageSent` event.
+MUST support global chain-ids. Propreitory chain ids should be mapped internally to global chain ids.
+MUST revert if `chain id` is invalid.
+MUST return `true` on successful response from underlying messaging protocol.
+
+**NOTE:** I'm proposing chain-ids be the bytes encoding of their native token name string. For eg., abi.encode("ETH"), abi.encode("SOL") imagining they cannot override.
+       
+       function sendMessage(bytes chainId, bytes receiver, bytes message) public returns(bool)
+
+receiveMessage
+Receive message/state from `sender` on a predetermined `chain id`
+MUST emit `MessageReceived` event.
+MUST support global based chain-ids.
+MUST revert if `sender` or `chain-id` is invalid.
+MUST return `true` on successful processing.
+
+**NOTE:** Sender validation (or) message validation, gas overrides are handled at relayer level.
+
+       function receiveMessage(bytes chainId, bytes sender, bytes message) public returns(bool)
+
+### Events
+
+MessageSent
+MUST trigger when message is sent, include zero bytes transfers.
+       
+       event MessageSent(bytes _sender, bytes _fromChainId, bytes _receiver, bytes _toChainId, bytes message)
+
+MessageReceived
+MUST trigger on any successful call to `receiveMessage(bytes chainId, bytes sender, bytes message)` function. 
+       
+       event MessageReceived(bytes _sender, bytes _fromChainId, bytes _receiver, bytes _toChainId, bytes message)
```

Comment:
The formatting here is missing. Would you mind e.g. putting this in a solidity interface with NatSpec comments?

Reply:
sure will do